### DISCUSSION TITLE
Validation after form submission.

### DISCUSSION THREAD

**Discussion by PlaygroundA** [react-hook-form/react-hook-form#3317] [Q&A] [answered]
Hello everyone :) I have a question. It may sound a little bit crazy, but still... Is there any way to prevent form validation in handleSubmit method? My problem: I have a field and it's validation depends on another field (SwitchButtonGroup). I have a custom async validation function and sometimes requests to my backend take a long time, so I'm trying to avoid unnecessary requests (implementing debounce and cancelToken (axios)). And I need to use 'onChange' mode to show validation errors after interaction with the form. But after clicking onSubmit, the form is validated again - which is normal behavior. To prevent this validation check I tried to play around formState.isSubmitting - unsuccessfully. Tried to save fieldValues in states and make some checks in validation function, but the logic gets complicated...and on top of that - I'm using fieldArrays. So....If you guys can help me with this question, I'd really appreciate it

Also want to ask about formState.isSubmitting.
I looked source code. Inside handleSubmit -`formState.isSubmitting : true`
But when I try `console.log(formState.isSubmitting) `inside validation function - after clicking onSubmit button, it's false.
I made super simple CSB. Please, open console and press onSubmit button.

[`https://codesandbox.io/s/react-hook-form-basic-forked-gwwdk?from-embed`](https://codesandbox.io/s/react-hook-form-basic-forked-gwwdk?from-embed)
Thanks a million

**Comment by bluebill1049** [comment, marked as answer]
I think perhaps you want to write a custom resolver for this behaviour? 
```
resolver:  asycn (data, context) => {
  console.log('you can have some context for the validation', context)
  if (data.xxx) {
     await xxxx
  }
}
```

**Reply by PlaygroundA** [reply]
It was my first attempt. But after a while, I had to give up this idea because had to do some common actions after successful/failed validation (set states and so on). So I wanted the component to encapsulate a validation function. 

P.S. Thanks for the answer

**Reply by bluebill1049** [reply]
or why not use validate function to include that aysnc validation?

**Reply by PlaygroundA** [reply]
Yes, I've already done this. I've updated CSB with Promise. If you have time, please, take a look. 

[https://codesandbox.io/s/react-hook-form-basic-forked-gwwdk?from-embed=&file=/src/index.js](https://codesandbox.io/s/react-hook-form-basic-forked-gwwdk?from-embed=&file=/src/index.js)

Open a console, print 'hello' - the validation function will return true and in console log you will see - "Name passed validation". 
Then after clicking on onSubmit button- this field will validating again - I don't want to send the request to the server in this step - I tried to check formState.isSubmitting inside validation function to prevent sending request if field has no errors and formState.isSubmitting = true, but this state always false inside this function. (if it makes any sense at all...)

Sorry, if I confuse you..

**Reply by bluebill1049** [reply]
try this: 

```tsx
const isSubmittingRef = React.useRef();

  isSubmittingRef.current = isSubmitting

  const asyncValidationFunction = (value) => {
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        console.log(isSubmittingRef.current)
        if (value === "hello") {
          resolve(true);
        } else {
          reject();
        }
      }, 500);
    });
  };
```

**Reply by PlaygroundA** [reply]
Yes, now it's true. Thanks a million for answering and for great library ! :)

**Reply by bluebill1049** [reply]
no worries buddy.
